Tests/MongoSwiftTests/CrudTests.swift

```diff
@@ -167,7 +167,7 @@ private class CrudTest {
         self.args = try operation.get("arguments")
         let outcome: Document = try test.get("outcome")
         self.error = outcome["error"] as? Bool
-        self.result = outcome["result"]
+        self.result = outcome["result"]!
```

Comment:
Is this forced unwrap only necessary because we're no longer using an optional for `CrudTest.result`?

The CRUD test's readme file still states:

> This will have some or all of the following fields:

So I would be included to leave `result` as an optional in case it doesn't exist. I could see that being the case for tests where `error` is true and a result is never returned nor asserted.

Reply:
Yup, that is the only reason for it being there. The decision on what to do here depends on the earlier conversation about `self.result`.

If "missing" values for fields in the JSON are `null` then we can just keep this as is otherwise it makes more sense to revert it back to an optional.

Reply:
The `null` in the JSON is technically asserting the absence of a Document for the various `findAndModify` helpers (as explained in [these CRUD spec doc blocks](https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#find-and-modify)). Since our helpers in `MongoCollection+FindAndModify` return `CollectionType?`, I think `null` here (even if parsed as NSNull since we interpret as a Document) should lead us to assert that those helpers return `nil`.

Reply:
I am not sure I follow, sorry: doesn't this suggest that we should just keep a non-optional `result` if we're expecting a true null anyways in the JSON? Otherwise, we'd have to make it optional but make sure we treat both `nil` and `NSNull` in the same way whenever we use `result` in any logic.

Correct me if I've misunderstood something -- I probably have.

Reply:
Right. If it is indeed the case that `"result"` may not exist sometime in the future, then we should revert this back to an optional. This is something I was hoping to get an answer for with [this comment](https://github.com/mongodb/mongo-swift-driver/pull/145#discussion_r235063481), so thanks!

I'll move over any new updates on this convo over to the earlier thread. Feel free to resolve this.